File: app/src/main/java/com/apanda/base/Module/Welcome/StartupActivity.java
package com.apanda.base.Module.Welcome;

import android.animation.Animator;
import android.animation.AnimatorSet;
import android.animation.ObjectAnimator;
import android.animation.PropertyValuesHolder;
import android.view.animation.AccelerateInterpolator;
import android.widget.TextView;

import com.apanda.base.Module.Gank.activity.GankActivity;
import com.apanda.base.R;
import com.apanda.base.base.BaseActivity;

import butterknife.Bind;

public class StartupActivity extends BaseActivity {
    @Bind(R.id.iv_logo)
    TextView _IvLogo;

    /**
     * 是否支持滑动返回
     *
     * @return
     */
    protected boolean supportSlideBack() {
        return false;
    }


    @Override
    public int getLayoutId() {
        return R.layout.activity_startup;
    }

    @Override
    public void initPresenter() {

    }

    @Override
    public void initView() {


        PropertyValuesHolder alpha = PropertyValuesHolder.ofFloat("alpha", 0.3f, 1f);
        PropertyValuesHolder scaleX = PropertyValuesHolder.ofFloat("scaleX", 0.3f, 1f);
        PropertyValuesHolder scaleY = PropertyValuesHolder.ofFloat("scaleY", 0.3f, 1f);
        ObjectAnimator objectAnimator1 = ObjectAnimator.ofPropertyValuesHolder(_IvLogo, alpha, scaleX, scaleY);
        ObjectAnimator objectAnimator2 = ObjectAnimator.ofPropertyValuesHolder(_IvLogo, alpha, scaleX, scaleY);

        AnimatorSet animatorSet = new AnimatorSet();
        animatorSet.playTogether(objectAnimator1, objectAnimator2);
        animatorSet.setInterpolator(new AccelerateInterpolator());
        animatorSet.setDuration(2000);
        animatorSet.addListener(new Animator.AnimatorListener() {
            @Override
            public void onAnimationStart(Animator animator) {

            }

            @Override
            public void onAnimationEnd(Animator animator) {
              startActivity(GankActivity.class);
                finish();
            }

            @Override
            public void onAnimationCancel(Animator animator) {

            }

            @Override
            public void onAnimationRepeat(Animator animator) {

            }
        });
        animatorSet.start();


    }

}

File: app/src/main/java/com/apanda/base/Module/Gank/activity/StartupActivity.kt
package com.apanda.base.Module.Gank.activity

import android.animation.Animator
import android.animation.AnimatorSet
import android.animation.ObjectAnimator
import android.animation.PropertyValuesHolder
import android.os.Message
import android.view.animation.AccelerateInterpolator
import android.widget.TextView
import butterknife.Bind
import com.apanda.base.R
import com.apanda.base.base.BaseActivity
import com.apanda.base.base.BaseModel
import com.apanda.base.base.BasePresenter

class StartupActivity : BaseActivity<BasePresenter<*, *>, BaseModel>(){
    @Bind(R.id.iv_logo)
    internal var _IvLogo: TextView? = null

    /**
     * 是否支持滑动返回

     * @return
     */
    override fun supportSlideBack(): Boolean {
        return false
    }


    override fun refresh(msg: Message) {

    }

    override val layoutId: Int
        get() = R.layout.activity_startup

    override fun initPresenter() {

    }

    override fun initView() {


        val alpha = PropertyValuesHolder.ofFloat("alpha", 0.3f, 1f)
        val scaleX = PropertyValuesHolder.ofFloat("scaleX", 0.3f, 1f)
        val scaleY = PropertyValuesHolder.ofFloat("scaleY", 0.3f, 1f)
        val objectAnimator1 = ObjectAnimator.ofPropertyValuesHolder(_IvLogo, alpha, scaleX, scaleY)
        val objectAnimator2 = ObjectAnimator.ofPropertyValuesHolder(_IvLogo, alpha, scaleX, scaleY)

        val animatorSet = AnimatorSet()
        animatorSet.playTogether(objectAnimator1, objectAnimator2)
        animatorSet.interpolator = AccelerateInterpolator()
        animatorSet.duration = 2000
        animatorSet.addListener(object : Animator.AnimatorListener {
            override fun onAnimationStart(animator: Animator) {

            }

            override fun onAnimationEnd(animator: Animator) {
                startActivity(HomeActivity::class.java)
                finish()
            }

            override fun onAnimationCancel(animator: Animator) {

            }

            override fun onAnimationRepeat(animator: Animator) {

            }
        })
        animatorSet.start()


    }

}
